app/controllers/receive_message_controller.rb

```diff
@@ -1,7 +1,10 @@
 class ReceiveMessageController < ApplicationController
   def receive
     from_number = params['From']
-    message     = User.new_conversation(from_number).next
+    user        = User.new(number: from_number)
+    message     = user.next_message
+
+    user.increment_conversation
```

Comment:
I think we should increment the  conversation "Before" fetching the message.
In this case the .increment_conversation is called "only" when a message from the user is received. and we are sending the "following message in the conversation"
Which will be the same that is sent in the remainder where "no increment" should occurr. since we dont know if the reminder was replied.